contracts/BinanceAdapter.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity 0.6.6;
+
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol"; 
+import "./IERC677.sol";
+
+contract BinanceAdapter {
+    event WithdrawToBinance(address indexed token, address indexed to, uint256 amountDatacoin, uint256 amountOtheroken);
+
+    IUniswapV2Router02 public honeyswapRouter;
+    address public bscBridge;
+    IERC677 public dataCoin;
+    address public convertToCoin;
+    //optional intermediate token for liquidity path
+    address public liquidityToken;
+    /*
+    ERC677 callback
+    */
+    function onTokenTransfer(address, uint256 amount, bytes calldata data) external returns (bool) {
+        uint256 maxint = uint256(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
+        //min output is 1 wei, no deadline
+        _withdrawToBinance(toAddress(data), amount, convertToCoin, 1, maxint);
```

Comment:
@hpihkala one thing to note here is that there's no min output. The amount of `convertToCoin` is just determined by market.

Reply:
Could we add a (read-only) helper function here to estimate the output?

Reply:
yes, I'll add. Also possible to encode a min return amount in `data` , but not sure how that would be passed thru DU2.

Reply:
Yeah we can't easily source that from the DU contract, so maybe we can leave that out of scope for now. The amounts withdrawn will generally be quite small in DU context, and if we assume there's reasonable liquidity available, and that attackers have little interest to front-run such low-value transactions, it shouldn't be much of a problem.